jabiseo-api/src/main/java/com/jabiseo/common/ValueOfEnumValidator.java

```diff
@@ -0,0 +1,35 @@
+package com.jabiseo.common;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class ValueOfEnumValidator implements ConstraintValidator<EnumValid, String> {
+
+    private EnumValid enumValid;
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        boolean result = false;
+        if(value == null){
+            return false;
+        }
+
+        Enum<?>[] enumValues = this.enumValid.enumClass().getEnumConstants();
+        if (enumValues != null) {
+            for (Object enumValue : enumValues) {
+                if (value.equals(enumValue.toString())
+                        || this.enumValid.ignoreCase() && value.equalsIgnoreCase(enumValue.toString())) {
+                    result = true;
+                    break;
+                }
+            }
+        }
+        return result;
```

Comment:
아래와 같이 짜봤는데 괜찮다면 반영해주셔도 좋을 것 같아요!~
1. result 변수 삭제
2. enumValues null 체크 early return
```suggestion
        if(value == null){
            return false;
        }

        Enum<?>[] enumValues = this.enumValid.enumClass().getEnumConstants();

        if (enumValues == null) {
            return false;
        }
        for (Object enumValue : enumValues) {
            if (value.equals(enumValue.toString())
                    || this.enumValid.ignoreCase() && value.equalsIgnoreCase(enumValue.toString())) {
                return true;
            }
        }
        return false;
```

Reply:
좋은 의견입니다 훨씬 depth가 줄어들었네요